### PR TITLE
add Tmp daily clean config on vmslaves

### PR DESCRIPTION
**Pull request by changev**
Thanks @sunnyqianzhang for pair programming!

### CODE REVIEW

**Comment by sunnyqianzhang** [comment]
This looks good to me +1

**Comment by sunnyqianzhang** [comment]
@anhou @linlynn Add cc for review.This PR is only used for documenting what we did to the vmslaves to clean the files under /tmp

**Comment by panpan0000** [comment]
PR itself is ok to me . 
if it can resolve current Ubuntu 14.04 tmp inode use up issue on Jenkins slaves, then it will be a +1

File: vmslave-config/ansible/roles/tmp-clean/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+- name: create tmpfiles.d folder
+  become: yes
+  shell: |
+    mkdir -p /etc/tmpfiles.d
+- name: put tmp.conf config file for ubuntu
+  become: yes
+  copy:
+    src: tmp.conf
+    dest: /etc/tmpfiles.d/tmp.confonrack@onrack:~/workspace/src/on-build-config/vmslave-config/ansible/roles
```
Comment (panpan0000) [inline vmslave-config/ansible/roles/tmp-clean/tasks/main.yml]: are you sure it's correct ?

**Comment by panpan0000** [comment]
@changev , please address the comments

**Comment by panpan0000** [comment]
can you please add RAC-ID and explain the technical solution to help better review?

File: vmslave-config/ansible/roles/tmp-clean/files/clear-ci-log [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+find /tmp -mtime +1 -name npm* -execdir rm -rf "{}" \;
+find /tmp -mtime +1 -name d2017* -execdir rm -rf "{}" \;
```
Comment (panpan0000) [inline vmslave-config/ansible/roles/tmp-clean/files/clear-ci-log]: what's d2017?

**Comment by sunnyqianzhang** [comment]
:+1:

**Comment by panpan0000** [comment]
👍 
this PR will address RAC-5271 ```/tmp is getting filled up with left-over files on vmslave15, vmslave16, and vmslave18```

File: vmslave-config/ansible/roles/tmp-clean/files/clear-ci-log [hunk 3]
```diff
@@ -0,0 +1,6 @@
+#!/bin/sh
+year_string=`date +%Y`
+year_log_pattern='d'${year_string}*
+find /tmp -mtime +1 -name ${year_log_pattern}
+find /tmp -mtime +1 -name npm* -execdir rm -rf "{}" \;
+find /tmp -atime +1 -name vagrant* -exec rm -rf "{}" \;
```
Comment (PengTian0) [inline vmslave-config/ansible/roles/tmp-clean/files/clear-ci-log:6]: Does the PR only clear directories and files whose name starts with "npm" or "vagrant"?
I just find there are also many directories whose name starts with "d", such as: d20170628-12047-1gp2yx4
Could you please also clear them?

Comment (changev) [reply vmslave-config/ansible/roles/tmp-clean/files/clear-ci-log:6]: @sunnyqianzhang

Comment (changev) [reply vmslave-config/ansible/roles/tmp-clean/files/clear-ci-log:6]:
```
year_log_pattern='d'${year_string}*
find /tmp -mtime +1 -name ${year_log_pattern}
```
I think these two lines of code will clear "d"

**Review by anhou — APPROVED** [review]

**Pull request merged**